src/views/home/components/transaction-form/transaction-form.view.tsx

```diff
@@ -41,42 +54,50 @@ const TransactionForm: FC<TransactionFormProps> = ({ onSubmit }) => {
   // };
 
   const onChainFromButtonClick = (from: Chain) => {
-    if (env && transactionData) {
-      const to = env.chains.find((chain) => chain.chainId !== from.chainId);
-
+    if (env && formData) {
+      const to = env.chains.find((chain) => chain.key !== from.key);
       if (to) {
-        setTransactionData({ ...transactionData, from, to });
+        setFormData({ ...formData, from, to, amount: undefined });
         setList(undefined);
       }
     }
   };
 
   const onInputChange = ({ amount, error }: { amount?: BigNumber; error?: string }) => {
-    if (transactionData && amount) {
-      setTransactionData({ ...transactionData, amount });
+    if (formData && amount) {
+      setFormData({ ...formData, amount });
       setError(error);
     }
   };
 
   const onFormSubmit = (e: React.FormEvent) => {
     e.preventDefault();
-    if (transactionData) {
-      onSubmit(transactionData);
+    if (formData && formData.amount) {
+      onSubmit({
+        ...formData,
+        amount: formData.amount,
+      });
     }
   };
 
   useEffect(() => {
-    if (env) {
-      setTransactionData({
+    if (formData?.from && account.status === "successful") {
+      void getBalance(formData.from.provider, account.data).then(setBalanceFrom);
+      void getBalance(formData.to.provider, account.data).then(setBalanceTo);
```

Comment:
When we make account a `string` and remove the function `getBalance` I'd remove the if and use `formData?.from.provider` twice:

```typescript
useEffect(() => {
  void formData?.from.provider.getBalance(account).then(setBalanceFrom);
  void formData?.to.provider.getBalance(account).then(setBalanceTo);
}, [account, formData?.from.provider, formData?.to.provider]);
```

It feels weird to test for `formData?.from` and not `formData?.to` in the if.

Reply:
I'm not buying account as string yet :laughing: